contracts/whitelist/src/contract.rs

```diff
@@ -0,0 +1,450 @@
+            max: MAX_PER_ADDRESS_LIMIT.to_string(),
+            got: msg.per_address_limit.to_string(),
+        });
+    }
+    if msg.per_address_limit == 0 {
+        return Err(ContractError::InvalidPerAddressLimit {
+            max: "must be > 0".to_string(),
+            got: msg.per_address_limit.to_string(),
+        });
+    }
+
+    let config = Config {
+        admin: info.sender.clone(),
+        start_time: msg.start_time,
+        end_time: msg.end_time,
+        num_members: msg.members.len() as u32,
+        unit_price: msg.unit_price,
+        per_address_limit: msg.per_address_limit,
+    };
+    CONFIG.save(deps.storage, &config)?;
+
+    if msg.start_time > msg.end_time {
+        return Err(ContractError::InvalidStartTime(
+            msg.start_time,
+            msg.end_time,
+        ));
+    }
+
+    if msg.start_time.is_expired(&env.block) {
+        return Err(ContractError::InvalidStartTime(
+            Expiration::AtTime(env.block.time),
+            msg.start_time,
+        ));
+    }
+
+    let fee_msgs = burn_and_distribute_fee(env, &info, CREATION_FEE)?;
+
+    if MAX_MEMBERS <= (config.num_members) {
+        return Err(ContractError::MembersExceeded {
+            expected: MAX_MEMBERS,
+            actual: config.num_members,
+        });
+    }
+
+    for member in msg.members.into_iter() {
+        let addr = deps.api.addr_validate(&member.clone())?;
+        WHITELIST.save(deps.storage, addr, &Empty {})?;
+    }
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("contract_name", CONTRACT_NAME)
+        .add_attribute("contract_version", CONTRACT_VERSION)
+        .add_attribute("sender", info.sender)
+        .add_messages(fee_msgs))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::UpdateStartTime(time) => execute_update_start_time(deps, env, info, time),
+        ExecuteMsg::UpdateEndTime(time) => execute_update_end_time(deps, env, info, time),
+        ExecuteMsg::UpdateMembers(msg) => execute_update_members(deps, env, info, msg),
+        ExecuteMsg::UpdatePerAddressLimit(per_address_limit) => {
+            execute_update_per_address_limit(deps, env, info, per_address_limit)
+        }
+    }
+}
+
+pub fn execute_update_start_time(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    start_time: Expiration,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    config.start_time = start_time;
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new()
+        .add_attribute("action", "update_start_time")
+        .add_attribute("start_time", start_time.to_string()))
+}
+
+pub fn execute_update_end_time(
```

Comment:
You can use generalized `execute_update_config` function with optional values like in https://github.com/deus-labs/junotools-contracts/blob/main/contracts/jt-airdrop-controller/src/contract.rs#L91.
Makes the code and msgs smaller. This is optional.